consensus/service/src/config/network.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Consensus network configuration.
+
+use mc_common::{HashMap, HashSet, NodeID, ResponderId};
+use mc_consensus_scp::{QuorumSet, QuorumSetMember};
+use mc_util_uri::{ConnectionUri, ConsensusPeerUri as PeerUri};
+use serde::{Deserialize, Serialize};
+use std::{fs, path::Path};
+
+/// Consensus network configuration.
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct NetworkConfig {
+    /// The set of nodes which you trust to validate transactions.
+    pub quorum_set: QuorumSet<ResponderId>,
+
+    /// List of peers we connect to.
+    pub broadcast_peers: Vec<PeerUri>,
+
+    /// List of URLs to use for transaction data.
+    pub tx_source_urls: Vec<String>,
+
+    /// Optional list of peers we are aware of.
+    pub known_peers: Option<Vec<PeerUri>>,
+}
+
+impl NetworkConfig {
+    /// Get the network configuration by loading the network.toml/json file.
+    pub fn load_from_path(path: impl AsRef<Path>, peer_responder_id: &ResponderId) -> Self {
```

Comment:
It seems worth documenting that these methods panic.

Reply:
I am planning to change them to not panic and return `Result<Self, String>` so that the caller can deal with it in one place. Didn't want to do it in this PR to reduce scope as much as possible.